canopen/objectdictionary/__init__.py

```diff
@@ -92,8 +94,92 @@ def add_object(self, obj):
         self.indices[obj.index] = obj
         self.names[obj.name] = obj
 
-
-class Record(collections.MutableMapping):
+    def get_object(self, index, subindex):
+        if index in self.indices:
+            obj = self.indices[index]
+        elif index in self.names:
+            obj = self.names[index]
+        else:
+            # Index does not exist
+            raise OdIndexError("Index {} is not present in the "
+                               "object dictionary".format(index))
+        if not isinstance(obj, Variable):
+            # Group or array
+            if subindex not in obj:
+                # Subindex does not exist
+                raise OdSubIndexError("Object at index {} does not have a "
+                                      "subindex {}".format(index, subindex))
+            obj = obj[subindex]
+        return obj
+
+
+class ValueStore(object):
+    """Provides a common interface to set and get all values out of object
+    types that hold a collection of values."""
+    subindices = {}
```

Comment:
They don't, because all classes inheriting from this class (**Record** and **Array**) define their private subindices dicitionary in their own `__init__` method, which takes precedence over the class variable. This works in this specialized case, but I agree that it is not directly obvious. If you prefer to make is explicit I can add an `__init__` method.

Reply:
Right, in that case you can keep it.